web-animations-1/Overview.bs

```diff
@@ -4810,8 +4827,15 @@ interface KeyframeEffect : AnimationEffect {
     This is `null` if the [=effect target=] is an {{Element}} or is itself `null`.
     When not `null`, this must be a [=pseudo-element=] selector with
     [[selectors-4#pseudo-element-syntax|valid syntax]].
-    This will not accept invalid values and will always return the selector in
-    its canonical form (i.e. `::before` instead of `:before`).
+
+    This will only accept `null` or syntactically-valid selectors
+    (i.e. `:before`, `:after`, `:first-letter`, `:first-line`, or a string starting with `::`).
+    If such a string is not given, this will throw a {{DOMException}} with
+    error name {{SyntaxError}}.
+
+    If one of the legacy single-colon selectors is given, the selector is
+    changed to its double-colon version. If the empth string is given,
```

Comment:
Ditto re being above the 'If such a string' line (also, 'empty' I presume :D)

Reply:
fixed. Also, are pseudo-element selectors case-sensitive. If not, we should canonicalize the case as well.

Reply:
It looks like it still says `the empth string`.

Could we rewrite this as something a bit more imperative like:

```
The [=target pseudo-selector=].

`null` if this effect has no [=effect target=] or
if the [=effect target=] is an element (i.e. not a pseudo-element).
When the [=effect target=] is a pseudo-element,
this specifies the pseudo-element selector (e.g. "::before").

On setting,
sets the [=target pseudo-selector=] of the [=animation effect=]
to the provided value after applying the following exceptions:

*   If the provided value is not a syntactically valid
    <pseudo-element-selector> the user agent must [=throw=]
    a {{DOMException}} with error name {{TypeError}} and leave the
    [=target pseudo-selector=] of this [=animation effect=] unchanged.

*   If one of the legacy Selectors Level 2 single-colon selectors
    (':before', ':after', ':first-letter', or ':first-line') is
    specified, the [=target pseudo-selector=] must be set to the
    equivalent two-colon selector (e.g. '::before').

*   If the empty string is specified, the [=target pseudo-selector=] is 
    cleared (hence getting this member subsequently would return
    `null`).

```

Although, now that I look at it, do we need that last condition?

Isn't it better to only allow _either_ `null` or an empty string, but not both?

Reply:
done